File: app/src/main/java/com/nipa/healthcareapp/PatientDashboardActivity.java
package com.nipa.healthcareapp;

import android.os.Bundle;
import android.view.View; // Required for FAB OnClickListener lambda
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.navigation.NavigationView;
import java.util.HashSet;
import java.util.Set;

public class PatientDashboardActivity extends AppCompatActivity {

    private AppBarConfiguration appBarConfiguration;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_patient_dashboard);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        DrawerLayout drawerLayout = findViewById(R.id.drawerLayout);
        NavigationView navView = findViewById(R.id.navView);
        // It's important that R.id.navHostFragment is the ID of a NavHostFragment in activity_patient_dashboard.xml
        NavController navController = Navigation.findNavController(this, R.id.navHostFragment);

        // Define top-level destinations from the Kotlin code
        Set<Integer> topLevelDestinations = new HashSet<>();
        topLevelDestinations.add(R.id.nav_providers);
        topLevelDestinations.add(R.id.nav_notifications);
        topLevelDestinations.add(R.id.nav_chats);
        topLevelDestinations.add(R.id.nav_upload);
        topLevelDestinations.add(R.id.nav_settings);

        appBarConfiguration = new AppBarConfiguration.Builder(topLevelDestinations)
                .setOpenableLayout(drawerLayout) // Correctly using setOpenableLayout for DrawerLayout
                .build();

        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
        NavigationUI.setupWithNavController(navView, navController);

        FloatingActionButton fabUpload = findViewById(R.id.fabUpload);
        fabUpload.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                navController.navigate(R.id.nav_upload);
            }
        });
    }

    @Override
    public boolean onSupportNavigateUp() {
        NavController navController = Navigation.findNavController(this, R.id.navHostFragment);
        return NavigationUI.navigateUp(navController, appBarConfiguration)
                || super.onSupportNavigateUp();
    }
}

File: app/src/main/java/com/nipa/healthcareapp/PatientDashboardActivity.kt
package com.nipa.healthcareapp

import android.content.Intent // Added Intent import
import android.os.Bundle
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import androidx.navigation.NavController
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import androidx.navigation.ui.NavigationUI.*;
import com.google.android.material.navigation.NavigationView
import com.nipa.healthcareapp.databinding.ActivityPatientDashboardBinding

class PatientDashboardActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    private lateinit var binding: ActivityPatientDashboardBinding
    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityPatientDashboardBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.toolbar)

        navController = findNavController(R.id.navHostFragment)

        appBarConfiguration = AppBarConfiguration(
            setOf(
                R.id.nav_providers,
                R.id.nav_notifications,
                R.id.nav_chats,
                R.id.nav_upload,
                R.id.nav_my_appointments, // Added new menu item
                R.id.nav_book_appointment,
                R.id.nav_settings,
                R.id.btnManagePatientProfile,
                R.id.nav_logout
            ),
            binding.drawerLayout
        )

        setupActionBarWithNavController(navController, appBarConfiguration)
        binding.navView.setupWithNavController(navController)
        binding.navView.setNavigationItemSelectedListener(this)

        binding.fabUpload.setOnClickListener { // Changed to use binding
            navController.navigate(R.id.nav_upload)
        }
        // The welcome message (tvPatientWelcome) is in nav_header_main.xml,
        // which is included by binding.navView. Its text is set via XML.
    }

    override fun onSupportNavigateUp(): Boolean {
        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
    }

    override fun onBackPressed() {
        if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
            binding.drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.btnManagePatientProfile -> {
                val intent = Intent(this, PatientProfileActivity::class.java)
                startActivity(intent)
                binding.drawerLayout.closeDrawer(GravityCompat.START) // Close the drawer
                return true // Item click handled
            }
            R.id.nav_book_appointment -> {
                startActivity(Intent(this, SelectProfessionalActivity::class.java))
                binding.drawerLayout.closeDrawer(GravityCompat.START)
                return true
            }
            R.id.nav_my_appointments -> {
                startActivity(Intent(this, PatientAppointmentsActivity::class.java))
                binding.drawerLayout.closeDrawer(GravityCompat.START)
                return true
            }
            // Potentially handle other items like nav_logout here if needed
            // For now, let NavController handle other items by default
        }
        // Default handling for other menu items by NavController
        // This ensures that if an item is in the nav graph, NavController will handle it.
        val navigated = item.onNavDestinationSelected(navController)
        if (navigated) {
            binding.drawerLayout.closeDrawer(GravityCompat.START)
        }
        return navigated || super.onOptionsItemSelected(item) // Fallback for items not in nav graph
    }
}
